src/test/java/com/smartbear/postman/script/PostmanScriptParserTest.java

```diff
@@ -38,27 +40,61 @@
     private WsdlProject project;
     private Assertable assertable;
 
+    List<TestAssertion> assertions;
+
+    GroovyScriptAssertion groovyAssertion;
+    ValidHttpStatusCodesAssertion validHttpStatusCodesAssertion;
+    InvalidHttpStatusCodesAssertion invalidHttpStatusCodesAssertion;
+    ResponseSLAAssertion responseSLAAssertion;
+    EqualsAssertion equalsAssertion;
+    SimpleContainsAssertion simpleContainsAssertion;
+
     @Before
     public void prepare() {
         project = new WsdlProject();
         assertable = mock(Assertable.class);
+
+        groovyAssertion = mock(GroovyScriptAssertion.class);
+        validHttpStatusCodesAssertion = mock(ValidHttpStatusCodesAssertion.class);
+        invalidHttpStatusCodesAssertion = mock(InvalidHttpStatusCodesAssertion.class);
+        responseSLAAssertion = mock(ResponseSLAAssertion.class);
+        equalsAssertion = mock(EqualsAssertion.class);
+        simpleContainsAssertion = mock(SimpleContainsAssertion.class);
+
+        when(assertable.addAssertion(GroovyScriptAssertion.LABEL)).thenReturn(groovyAssertion);
+        when(assertable.addAssertion(ValidHttpStatusCodesAssertion.LABEL)).thenReturn(validHttpStatusCodesAssertion);
+        when(assertable.addAssertion(InvalidHttpStatusCodesAssertion.LABEL)).thenReturn(invalidHttpStatusCodesAssertion);
+        when(assertable.addAssertion(ResponseSLAAssertion.LABEL)).thenReturn(responseSLAAssertion);
+        when(assertable.addAssertion(EqualsAssertion.LABEL)).thenReturn(equalsAssertion);
+        when(assertable.addAssertion(SimpleContainsAssertion.LABEL)).thenReturn(simpleContainsAssertion);
+
+        assertions = new ArrayList<>();
+        when(assertable.getAssertionList()).thenReturn(assertions);
+    }
+
+    @Test
+    public void newLineTokenIsFirst() {
+        //???
+        //Each token regular expression appended in the beginning with "\s*" pattern which includes new line pattern,
+        //so the NEW_LINE pattern must be the first in the pattern check list, so it must be the first in the enum.
+        assertTrue("The NEW_LINE pattern must be the first in the enum TokenType", TokenType.values()[0] == TokenType.NEW_LINE);
```

Comment:
this looks very suspecios to use order inside the enum

Reply:
The parser is finite state machine. And it's splitted by several finite state machines (regexps) with the specific order.